src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/RazorSyntaxNodeExtensions.cs

```diff
@@ -5,17 +5,24 @@
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using Microsoft.AspNetCore.Razor;
 using Microsoft.AspNetCore.Razor.Language;
 using Microsoft.AspNetCore.Razor.Language.Syntax;
-using Microsoft.AspNetCore.Razor.LanguageServer.Formatting;
 using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Range = Microsoft.VisualStudio.LanguageServer.Protocol.Range;
 
-namespace Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
+namespace Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
 
-internal static class SyntaxNodeExtensions
+#pragma warning disable IDE0065 // Misplaced using directive
```

Comment:
@CyrusNajmabadi should I file a bug for this? These using directives cannot move outside the `namespace`, or the code will stop compiling.

Reply:
The rule is purely syntactic.  It's there reason they can't move outside?  WTH the inner affected code being written to not depend on that?

Reply:
They don't resolve correctly if moved outside. The `SyntaxNode/Token/Kind` from the nearest namespace is preferred.

Reply:
Yeah. Either you need to disable that rule, or update those cases to fully qualify to the right place. 

The rule is not "you can be on the wrong side if it would make the code uglier", it's firmly "you must be on the right side, no exceptions". So just disable it here :-)

Reply:
Right, I did. But that's why I was asking if it was considered a bug or not :slightly_smiling_face:.